_summaries/tancik2020fourier_1.md

```diff
@@ -1,19 +1,91 @@
+
+$$\gamma(\textbf{v}) = \[a_1 cos(2\pi \textbf{b}_1^T\textbf{v}), a_1 sin(2\pi \textbf{b}_1^T\textbf{v}), \dots, a_m cos(2\pi \textbf{b}_m^T\textbf{v}), a_m sin(2\pi \textbf{b}_m^T\textbf{v})\]^T$$
+
+The corresponding kernel function, using the property $$cos(\alpha)cos(\beta)+sin(\alpha)sin(\beta)=cos(\alpha -\beta)$$, then simplifies to:
+
+$$k_{\gamma}(\textbf{v}_1, \textbf{v}_2) = \gamma(\textbf{v}_1)^T \gamma(\textbf{v}_2) = \sum_{j=1}^{m}a_j^2 cos(2\pi \textbf{b}_j^T(\textbf{v}_1-\textbf{v}_2)) = h_{\gamma}(\textbf{v}_1-\textbf{v}_2)$$
+
+$$\text{where}\quad h_{\gamma}(\textbf{v}_{\Delta}) \triangleq \sum_{j=1}^{m}a_j^2 cos(2\pi \textbf{b}_j^T \textbf{v}_{\Delta})$$.
+
+Note that the kernel above is stationary, i.e. it only depends on the difference of the inputs. This is a Fourier approximation of the kernel function with $$a_j^2$$ and $$\textbf{b}_j$$ as the Fourier series coefficients and the Fourier basis frequencies respectively. The spectrum of the kernel can therefore be controlled using the same parameters. 
+
+Using these input embeddings to train a network is equivalent to kernel regression with the stationary NTK function. Thus, the training dynamics of the network, specifically the rate of convergence and the generalizability of the model can be tuned using the $$a_j$$ and $$\textbf{b}_j$$ parameters of the kernel to control the bandwidth of NTK as shown in figure below.
+
+<p align="center">
+  <img width="800" alt="tancik2020fourier_1b" src="tancik2020fourier_1b.png">
+</p>
+
+The magnitude of the higher frequency components reduces with the parameter $$p$$ in the figure. $$p=0$$ corresponds to equal magnitude of all frequencies and $$p=\infty$$ corresponds to just 1 frequency with non-zero magnitude. Figures (b) and (d) show that $$p=\infty$$ leads to underfitting and $$p=0$$ leads to overfitting. Therefore, $$p=1$$ has adequate power across frequencies to still learn higher frequency functions rapidly but not overfit them. Figure (c) shows the convergence of the different frequency components of the training error. 
+
+
+### Application of Fourier feature mapping
+
+In real-world problems of higher dimensions, it is not possible to densely sample the Fourier basis frequencies for the feature mapping because of scalability issues. Therefore, the paper proposes to sample a smaller set of random Fourier features from a parametric distribution which matches the performance of the dense set of Fourier features. It further finds that the performance does not depend so much on the family of the distribution (distribution shape) but on its scale (standard deviation). Proved empirically in the figure below. The three subfigures are experiments with three different synthetic datasets.
+
```

Comment:
You can maybe add a line here to say, among the ones they tested Gaussian provided better results among a wider range of standard deviations.

Reply:
You mean "Gaussian provided better results among a wider range of distributions" or "Gaussian provided better results over a wider range of standard deviation"? And where does the paper say any of the two?